wom/src/main/scala/wom/types/WomFloatType.scala

```diff
@@ -13,9 +14,12 @@ case object WomFloatType extends WomPrimitiveType {
     case d: Double => WomFloat(d)
     case n: JsNumber => WomFloat(n.value.doubleValue())
     case f: WomFloat => f
-    case s: String => WomFloat(s.toDouble)
-    case s: JsString => WomFloat(s.value.toDouble)
-    case s: WomString => WomFloat(s.value.toDouble)
+    case s: String => WomFloat(s.asDouble)
+    case s: JsString => WomFloat(s.value.asDouble)
+    case s: WomString => WomFloat(s.value.asDouble)
+    case WomOptionalValue(WomFloatType, Some(d)) =>
+      println(s"My stars! That was a filled optional Float you just converted to a Float!")
```

Comment:
didn't actually need the filled-optional thing for floats yet but as long as I was in here...